src/Http/Http.Results/src/Results.cs

```diff
@@ -201,24 +201,32 @@ public static IResult Json(object? data, JsonSerializerOptions? options = null,
     /// recreating cached data with each call.</remarks>
 #pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
     public static IResult Json(object? data, JsonTypeInfo jsonTypeInfo, string? contentType = null, int? statusCode = null)
-#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
-        => Json<object>(data, jsonTypeInfo, contentType, statusCode);
+    {
+        ArgumentNullException.ThrowIfNull(jsonTypeInfo);
+        return new JsonHttpResult<object>(data, statusCode, contentType) { JsonTypeInfo = jsonTypeInfo };
+    }
 
     /// <summary>
     /// Creates a <see cref="IResult"/> that serializes the specified <paramref name="data"/> object to JSON.
     /// </summary>
     /// <param name="data">The object to write as JSON.</param>
-    /// <param name="jsonSerializerContext">Metadata about the type to convert.</param>
+    /// <param name="type">The type of object to write.</param>
+    /// <param name="context">A metadata provider for serializable types.</param>
     /// <param name="contentType">The content-type to set on the response.</param>
     /// <param name="statusCode">The status code to set on the response.</param>
     /// <returns>The created <see cref="JsonHttpResult{TValue}"/> that serializes the specified <paramref name="data"/>
     /// as JSON format for the response.</returns>
     /// <remarks>Callers should cache an instance of serializer settings to avoid
     /// recreating cached data with each call.</remarks>
 #pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
-    public static IResult Json(object? data, JsonSerializerContext jsonSerializerContext, string? contentType = null, int? statusCode = null)
-#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
-        => Json<object>(data, jsonSerializerContext, contentType, statusCode);
+    public static IResult Json(object? data, Type type, JsonSerializerContext context, string? contentType = null, int? statusCode = null)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+        return new JsonHttpResult<object>(data, statusCode, contentType)
+        {
+            JsonTypeInfo = context.GetTypeInfo(type) ?? throw new InvalidOperationException($"Unable to obtain the JsonTypeInfo for type '{type.FullName}' from the context '{context.GetType().FullName}'.")
```

Comment:
Add a `GetRequiredTypeInfo` helper method that centralizes the error and its message.